src/azure-cli/azure/cli/command_modules/vm/_actions.py

```diff
@@ -227,13 +253,21 @@ def _create_image_instance(publisher, offer, sku, version):
     }
 
 
-def _get_latest_image_version(cli_ctx, location, publisher, offer, sku):
-    top_one = _compute_client_factory(cli_ctx).virtual_machine_images.list(location,
-                                                                           publisher,
-                                                                           offer,
-                                                                           sku,
-                                                                           top=1,
-                                                                           orderby='name desc')
-    if not top_one:
-        raise CLIError("Can't resolve the version of '{}:{}:{}'".format(publisher, offer, sku))
+def _get_latest_image_version(cli_ctx, location, publisher, offer, sku, edge_zone=None):
+    if edge_zone is not None:
+        from azure.cli.core.commands.client_factory import get_mgmt_service_client
+        from azure.cli.core.profiles import ResourceType
+        edge_zone_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_COMPUTE).virtual_machine_images_edge_zone
+        top_one = edge_zone_client.list(location, edge_zone, publisher, offer, sku, top=1, orderby='name desc')
+        if not top_one:
+            raise CLIError("Can't resolve the version of '{}:{}:{}'".format(publisher, offer, sku))
```

Comment:
This judgment can be written in the outer layer, so as to avoid repeating it again

Reply:
Please use the specific error type `InvalidArgumentValueError` instead of `CLIError`

Reply:
> This judgment can be written in the outer layer, so as to avoid repeating it again

But it’s not the same format as above